engine/src/main/java/org/terasology/physics/bullet/EntityMotionState.java

```diff
@@ -16,47 +16,55 @@
 
 package org.terasology.physics.bullet;
 
-import com.bulletphysics.linearmath.MotionState;
-import com.bulletphysics.linearmath.Transform;
+import com.badlogic.gdx.physics.bullet.linearmath.btMotionState;
+import org.joml.Matrix4f;
+import org.joml.Quaternionf;
+import org.joml.Vector3f;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.logic.location.LocationComponent;
-import org.terasology.math.VecMath;
+import org.terasology.math.JomlUtil;
 
 /**
  * This motion state is used to connect rigid body entities to their rigid body in the bullet physics engine.
  * Bullet reads the initial state of the rigid body out of the entity, and then updates its location and rotation
  * as it moves under physics.
  *
  */
-public class EntityMotionState extends MotionState {
+public class EntityMotionState extends btMotionState {
+    private static final Logger logger = LoggerFactory.getLogger(EntityMotionState.class);
+
     private EntityRef entity;
+    private Quaternionf rot = new Quaternionf();
+    private Vector3f position = new Vector3f();
 
     /**
      * Only the BulletPhysics class is expected to create instances.
      *
-     * @param entity The entity to relate this motion state to and set the
-     *               LocationComponent of.
+     * @param entity The entity to relate this motion state to and set the LocationComponent of.
      */
     EntityMotionState(EntityRef entity) {
+        super();
         this.entity = entity;
     }
 
     @Override
-    public Transform getWorldTransform(Transform transform) {
+    public void getWorldTransform(Matrix4f transform) {
```

Comment:
Are we using returns  somewhere? in modules?
getWorldTransform with returns void - sounds strange..

Reply:
this syncs changes between the LocationComponent and the bullet rigid body. this is for the JNI layer and its not exposed to engine.